engine/evarutil.mli

```diff
@@ -199,6 +199,20 @@ val nf_evar_map_universes : evar_map -> evar_map * (Constr.constr -> Constr.cons
 exception Uninstantiated_evar of Evar.t
 val flush_and_check_evars :  evar_map -> constr -> Constr.constr
 
+(** [finalize env sigma f] combines universe minimisation,
+   evar-and-universe normalisation and universe restriction.
+
+    It minimizes universes in [sigma], calls [f] with [to_constr]
+   applied to the updated [sigma] and restricts the local universes of
+   [sigma] to those encountered while running [f].
+
+    Ideally we would instead have [f : unit -[Normalize : econstr ->
+   constr]-> 'a] ie explicit effects but our OCaml doesn't support it
+   yet. *)
+val finalize : ?abort_on_undefined_evars:bool -> env -> evar_map ->
```

Comment:
I think the sentence "Ideally.." is not clear and does not help the user. If `-[fobar]->` has no meaning in OCaml why are you using it here, what does it mean?

What the comment should state is that the `~to_constr` function (that I would label) *holds an imperative state in its closure*, hence it should not be used in code that does backtracking, e.g. try-with, since every time you call it it stashes data and you have no way to purge it.

Reply:
In other words I don't expect plugin writers to know what a type system with effects is, esp that these are not very widespread these days in mainstream programming languages.

For the rest of the PR, many thanks!

Reply:
Changed the doc a bit. I'm not quite satisfied with the wording so feel free to suggest changes / alternatives.

>that I would label

Why?

Reply:
It helps reading the type, IMHO.